File: ios/Sources/AdoboNetworkSocialLoginCapacitorPlugin/AdoboNetworkSocialLoginCapacitorPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AdoboNetworkSocialLoginCapacitorPlugin)
public class AdoboNetworkSocialLoginCapacitorPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AdoboNetworkSocialLoginCapacitorPlugin"
    public let jsName = "AdoboNetworkSocialLoginCapacitor"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = AdoboNetworkSocialLoginCapacitor()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/adobocorp/adobonetwork/sociallogincapacitor/AdoboNetworkSocialLoginCapacitorPlugin.java
package com.adobocorp.adobonetwork.sociallogincapacitor;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "AdoboNetworkSocialLoginCapacitor")
public class AdoboNetworkSocialLoginCapacitorPlugin extends Plugin {

    private AdoboNetworkSocialLoginCapacitor implementation = new AdoboNetworkSocialLoginCapacitor();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}
